core/src/main/java/nl/nn/adapterframework/filesystem/FtpFileSystem.java

```diff
@@ -281,6 +281,12 @@ public String getName(FTPFile file) {
 		}
 		return file.getName();
 	}
+	
+	@Override
+	public String getParentFolder(FTPFile file) throws FileSystemException {
+		FTPFileRef fileRef = (FTPFileRef) file;
+		return fileRef.getFolder();
```

Comment:
I don't think you can just cast this.

Reply:
before this method is called with file toFile method is called and that returns FtpFileRef but I agree this is confusing